common/src/disk.rs

```diff
@@ -292,13 +292,7 @@ impl FromStr for CompressionAlgorithm {
     PartialOrd,
     Ord,
 )]
-pub struct DatasetConfig {
-    /// The UUID of the dataset being requested
-    pub id: DatasetUuid,
-
-    /// The dataset's name
-    pub name: DatasetName,
-
+pub struct SharedDatasetConfig {
```

Comment:
This is used here, and also in the `NestedDatasetConfig` struct in `sled-storage`.

That struct is fine being internal to `sled-storage` only though, it's not crossing an API boundary.